OpenRA.Mods.Common/Traits/Mobile.cs

```diff
@@ -308,6 +308,12 @@ bool IsBlockedBy(Actor self, Actor otherActor, Actor ignoreActor, CellConditions
 			if (self == null)
 				return true;
 
+			// If self is structure producing units we are not blocked
+			if (self.TraitsImplementing<Production>().Count() > 0)
```

Comment:
Wouldn't Any() be faster?

Reply:
Maybe, but property, like length,  would work even better.

Reply:
This check is hiding the real bug, which is at
https://github.com/OpenRA/OpenRA/blob/d967c564a2dec515ef2afec9ca2790ef6b4381a5/OpenRA.Mods.Common/Traits/Production.cs#L148-L156

Somewhere down the call chain from `self.NotifyBlocker` it is assuming that the caller is Mobile, and somehow bypassing the normal trait system to call this method despite not having the required trait.

Fixing this might be worth a PR of its own.